---
title: "figure5"
author: "Jaime So"
date: "2022-07-19"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
library(reshape2)
library(tidyverse)
library(IRanges)
library(viridis)
library(colourvalues)
library(cowplot)
library(rstatix)
knitr::opts_chunk$set(fig.width=12, fig.height=8) 
```

# Assess genomic coverage of various Tbg isolates to VSG expressed in patients
Run bowtie using a variety of read lengths from sequenced T.b. gambiense genomes and calculate/compare coverage of assembled patient VSG from our study.

## Commands used to run analysis
```{r, eval=FALSE}

# raw sequencing read files each uploaded to their own directory designated by strain name
# datasets provided as paired end reads concatenated into single fq file and analyzed as if single end
# we can iterate through each genome using their distinctive directory names with this bash script

for path in seqFiles/*; do
        SAMPLEID="$(basename "${path}")"
# quality and adapter trim raw reads
trim_galore --dont_gzip -o seqFiles/$SAMPLEID/ seqFiles/$SAMPLEID/*.fq

# truncate reads to desired lengths
trimmomatic SE -threads 4 seqFiles/$SAMPLEID/*_trimmed.fq seqFiles/$SAMPLEID/20BP.fq CROP:20
trimmomatic SE -threads 4 seqFiles/$SAMPLEID/*_trimmed.fq seqFiles/$SAMPLEID/30BP.fq CROP:30
trimmomatic SE -threads 4 seqFiles/$SAMPLEID/*_trimmed.fq seqFiles/$SAMPLEID/50BP.fq CROP:50

# generated an indexed bowtie reference for Tbg Patient VSG called tbgPatientsbt
# run bowtie for each read query size
(bowtie -a --best -v 2 -S tbgPatientsbt seqFiles/$SAMPLEID/20BP.fq seqFiles/$SAMPLEID/20BP.bt.sam) 2> seqFiles/$SAMPLEID/20BPbowtieSE.txt
samtools view -bS seqFiles/$SAMPLEID/20BP.bt.sam -o seqFiles/$SAMPLEID/20BP.bt.bam
bedtools genomecov -ibam seqFiles/$SAMPLEID/20BP.bt.bam -bga > seqFiles/$SAMPLEID/genomecov.20BP.txt

(bowtie -a --best -v 2 -S tbgPatientsbt seqFiles/$SAMPLEID/30BP.fq seqFiles/$SAMPLEID/30BP.bt.sam) 2> seqFiles/$SAMPLEID/30BPbowtieSE.txt
samtools view -bS seqFiles/$SAMPLEID/30BP.bt.sam -o seqFiles/$SAMPLEID/30BP.bt.bam
bedtools genomecov -ibam seqFiles/$SAMPLEID/30BP.bt.bam -bga > seqFiles/$SAMPLEID/genomecov.30BP.txt

(bowtie -a --best -v 2 -S tbgPatientsbt seqFiles/$SAMPLEID/50BP.fq seqFiles/$SAMPLEID/50BP.bt.sam) 2> seqFseqFiles/$SAMPLEID/50BPbowtieSE.txt
samtools view -bS seqFiles/$SAMPLEID/50BP.bt.sam -o seqFiles/$SAMPLEID/50BP.bt.bam
bedtools genomecov -ibam seqFiles/$SAMPLEID/50BP.bt.bam -bga > seqFiles/$SAMPLEID/genomecov.50BP.txt

done

```

## Functions for importing bedtools genomecov data
Use iRanges R package to collapse overlapping reads into single range of VSG gene coverage
```{r, include=FALSE}

# function for reading and summarizing bowtie SE reports
bt_align <- function(directory){
  files <- list.files(path = directory, full.names = T)
  btdf <- data.frame(reads_processed = double(),
                     reads_aligned = double(),
                     unaligned = double(),
                     sample = character())
  for (i in 1:length(files)){
    test <- read_delim(file = files[i], delim = ":", col_names = F, comment = "(") 
    newdata <- data.frame(reads_processed = test[1,2],
                          reads_aligned = test[2,2],
                          unaligned = test[3,2],
                          sample = files[i])
    colnames(newdata) <- c("reads_processed", "reads_aligned", "unaligned", "sample")
    btdf <- rbind(btdf, newdata)
  }
  return(btdf)
}


# function for reading and summarizing bedtools genomecov output
read_genomecov <- function(inputname){
  x <- read_tsv(inputname, col_names = F)
  colnames(x) <- c("VSG", "start", "end", "depth")
  y <- x %>% group_by(VSG) %>% summarize(seqlen = max(end))
  # use iranges package for visualization
  plotR <- data.frame(start = double(),
                      end = double(),
                      width = double(),
                      VSG = character())
  for (i in 1:length(y$VSG)) {
    irR <- IRanges(x[x$depth >= 1, ] %>%
                   .[.$VSG == y$VSG[i], ] %>%
                   select(VSG, start, end) %>%
                   distinct() %>%
                   .$start,
                 end = x[x$depth >= 1, ] %>%
                   .[.$VSG == y$VSG[i], ] %>%
                   select(VSG, start, end) %>%
                   distinct() %>%
                   .$end)
    # collapse overlapping regions to calculate VSG coverage
    cov <- IRanges::reduce(irR) %>% as.data.frame()
    cov <- cov %>% mutate(VSG = y$VSG[i])
    plotR <- rbind(plotR, cov)
    }
  plotR$sample <- inputname
  plotR <- merge(plotR, y)
  return(plotR)
}
# function works, gives ranges per patient VSG


# pull and quantify VSG where there were no hits over the full VSG length
nohits <- function(inputname){
  x <- read_tsv(inputname, col_names = F)
  colnames(x) <- c("VSG", "start", "end", "depth")
  y <- x %>% group_by(VSG) %>% summarize(seqlen = max(end))
  x <- merge(x, y)
  x$sample <- inputname
  x <- x[x$depth == 0 & x$start == 0 & x$end == x$seqlen, ]
  return(x)
}

```

## Upload Data

```{r, include=FALSE}

# replace trinity gene name with cluster name
cluster_ref <- read_csv("FigureData/AllHATpatients_clusterReference.csv")

#20bp reads
nohits20bp <- list.files(path = "FigureData/20bpReads/", pattern = "genomecov", full.names = T) %>%
  map_df(~nohits(.)) %>%
  mutate(genome = sub("FigureData/20bpReads/", "", sample) %>% sub(".genomecov.20BP.txt", "", .)) %>%
  merge(., cluster_ref, by = "VSG")

bt20bp <- list.files(path = "FigureData/20bpReads",  pattern = "genomecov", full.names = T) %>%
  map_df(~read_genomecov(.)) %>%
  mutate(genome = sub("FigureData/20bpReads/", "", sample) %>% sub(".genomecov.20BP.txt", "", .)) %>%
  merge(., cluster_ref, by = "VSG")

#30bp reads
nohits30bp <- list.files(path = "FigureData/30bpReads", pattern = "genomecov", full.names = T) %>%
  map_df(~nohits(.)) %>%
  mutate(genome = sub("FigureData/30bpReads/", "", sample) %>% sub(".genomecov.30BP.txt", "", .)) %>%
  merge(., cluster_ref, by = "VSG")

bt30bp <- list.files(path = "FigureData/30bpReads",  pattern = "genomecov", full.names = T) %>%
  map_df(~read_genomecov(.)) %>%
  mutate(genome = sub("FigureData/30bpReads/", "", sample) %>% sub(".genomecov.30BP.txt", "", .)) %>%
  merge(., cluster_ref, by = "VSG")

#50bp reads
nohits50bp <- list.files(path = "FigureData/50bpReads", pattern = "genomecov", full.names = T) %>%
  map_df(~nohits(.)) %>%
  mutate(genome = sub("FigureData/50bpReads/", "", sample) %>% sub(".genomecov.50BP.txt", "", .)) %>%
  merge(., cluster_ref, by = "VSG")

bt50bp <- list.files(path = "FigureData/50bpReads",  pattern = "genomecov", full.names = T) %>%
  map_df(~read_genomecov(.)) %>%
  mutate(genome = sub("FigureData/50bpReads/", "", sample) %>% sub(".genomecov.50BP.txt", "", .)) %>%
  merge(., cluster_ref, by = "VSG")

# Import genome dataset metadata
genomeMeta <- read_csv(file = "FigureData/GenomeMappingMetadata.csv", col_names = T) 

```

## Analyze

```{r}
# calculate percent coverage
calc_cov <- function(df){
  pct_cov <- df %>% group_by(genome, cluster) %>% 
    summarize(aligned = sum(width)) %>% 
    full_join(df %>% select(cluster, seqlen) %>% distinct()) %>% 
    mutate(pct_coverage = (aligned / seqlen)*100)
  return(pct_cov)
}

# merge with all the VSG with 0 hits

calc_cov_20bp <- calc_cov(bt20bp)
# no VSG had 0 coverage with 20bp reads
calc_cov_30bp <- calc_cov(bt30bp)
calc_cov_30bp <- rbind(calc_cov_30bp, nohits30bp %>% 
                         mutate(aligned = 0) %>% 
                         mutate(pct_coverage = 0) %>% 
                         select(genome, cluster, aligned, seqlen, pct_coverage))

calc_cov_50bp <- calc_cov(bt50bp)
calc_cov_50bp <- rbind(calc_cov_50bp, nohits50bp %>% 
                         mutate(aligned = 0) %>% 
                         mutate(pct_coverage = 0) %>% 
                         select(genome, cluster, aligned, seqlen, pct_coverage))
```

## find patient N-term domain boundaries as determined by HMMscan pipeline
### compare coverage by domain
```{r}
# focus on 30bp set for more in-depth analysis
Nbound <- read_csv("FigureData/AllPatient_orf_VSGs_merged-translated_NtermSummary.csv") %>% select(original_seqID, nterm_seq_length) %>% mutate(nterm_coord = nterm_seq_length*3)
colnames(Nbound)[1] <- "VSG"

plot30bp <- merge(bt30bp, genomeMeta)
plot30bp <- left_join(plot30bp, Nbound, by = 'VSG')

# pick up any ranges that span the C-term/N-term boundary
DomSpans <-plot30bp[plot30bp$start <= plot30bp$nterm_coord & plot30bp$end >= plot30bp$nterm_coord, ] %>%
  select(cluster, genome, start, end, nterm_coord, seqlen)

addtoNterm <- DomSpans %>% select(cluster, genome, start, nterm_coord) %>% dplyr::rename(end = nterm_coord) %>% mutate(width = (end - start)+1)

addtoCterm <- DomSpans %>% select(cluster, genome,nterm_coord, end) %>% dplyr::rename(start = nterm_coord) %>% mutate(width = (end - start)+1)

Nterm <- plot30bp[plot30bp$end <= plot30bp$nterm_coord, ] %>% 
  select(cluster, genome, start, end, width) %>%
  rbind(.,addtoNterm) %>%
  distinct() %>%
  group_by(cluster, genome) %>% 
  summarize(aligned = sum(width)) %>% 
  merge(plot30bp %>% select(cluster, nterm_coord) %>% distinct()) %>% 
  mutate(pct_coverage = (aligned / nterm_coord)*100) %>%
  select(cluster, genome, pct_coverage, aligned, nterm_coord) %>%
  dplyr::rename(domain_length = nterm_coord)

# make entries for VSG with no hits
Nterm %>% group_by(genome) %>% summarise(n = n())

patientVSG <- unique(bt30bp$cluster)
genomes <- unique(bt30bp$genome)

add0Nhit <- data.frame(cluster = character(),
                       genome = character(),
                       pct_coverage = double())
for (i in 1:37){
  y <- data.frame(cluster = setdiff(patientVSG, Nterm[Nterm$genome == genomes[i], ] %>% .$cluster)) %>%
    mutate(genome = genomes[i]) %>%
    mutate(pct_coverage = 0, aligned = 0)
  add0Nhit <- rbind(add0Nhit, y)
}

add0Nhit <- inner_join(add0Nhit, plot30bp %>% select(cluster, nterm_coord)) %>% distinct() %>% dplyr::rename(domain_length = nterm_coord)

Nterm <- rbind(Nterm, add0Nhit)
Nterm %>% group_by(genome) %>% summarise(n = n())


Cterm <- plot30bp[plot30bp$start >= plot30bp$nterm_coord, ] %>% 
  select(cluster, genome, start, end, width) %>%
  rbind(.,addtoCterm) %>%
  distinct() %>%
  group_by(cluster, genome) %>% 
  summarize(aligned = sum(width)) %>% 
  merge(plot30bp %>% select(cluster, nterm_coord, seqlen) %>% distinct()) %>%
  mutate(pct_coverage = (aligned / (seqlen - nterm_coord + 1))*100, domain_length = (seqlen - nterm_coord + 1)) %>%
  select(cluster, genome, pct_coverage, aligned, domain_length)

# make entries for VSG with no hits
Cterm %>% group_by(genome) %>% summarise(n = n())

add0Chit <- data.frame(cluster = character(),
                       genome = character(),
                       pct_coverage = double())
for (i in 1:37){
  y <- data.frame(cluster = setdiff(patientVSG, Cterm[Cterm$genome == genomes[i], ] %>% .$cluster)) %>%
    mutate(genome = genomes[i]) %>%
    mutate(pct_coverage = 0, aligned = 0)
  add0Chit <- rbind(add0Chit, y)
}

add0Chit <- inner_join(add0Chit, plot30bp %>% select(cluster, nterm_coord, seqlen)) %>% mutate(domain_length = seqlen - nterm_coord +1) %>% distinct()
add0Chit$domain_length <- replace(add0Chit$domain_length, add0Chit$domain_length<0, 0)

Cterm <- rbind(Cterm, add0Chit %>% select(-nterm_coord,-seqlen))
Cterm %>% group_by(genome) %>% summarise(n = n())

```

### Fig5A: plot N-term/C-term domain coverage
Pick a representative set for ease of viewing (141BT, BRAZO, B4_4163P), we can see from supplemental figure 8A that most datasets tend to follow similar distribution

```{r}

distDomains <- rbind(Cterm %>% mutate(domain = 'C-Terminal Domain'), Nterm %>% mutate(domain = 'N-Terminal Domain')) %>%
  convert_as_factor(domain, genome) %>%
  merge(plot30bp %>% select(origin, year, genome) %>% distinct()) %>%
  ungroup()

distDomains %>% mutate(groups = case_when(year < "1999" ~ "Cote d'Ivoire 1980's",
                                               year > "2004" ~ "DRC 2000's",
                                               TRUE ~ "Cote d'Ivoire 2000's")) %>%
  .[.$genome == "141BT" | .$genome == "BRAZO" | .$genome == "B4_4163P", ] %>%
  ggplot(aes(x = pct_coverage, fill = groups)) +
  geom_density(alpha = 0.4) +
  scale_color_viridis(discrete = T) +
  scale_fill_viridis(discrete = T) +
  xlab("Percent Patient VSG Gene Coverage") +
  ggtitle("Domain Coverage Distribution") +
  facet_wrap(~domain, ncol = 1, scales = "free_y") +
  scale_x_continuous(limits = c(0,100)) +
  theme_bw() 

```

### Fig5B: dotplots of VSG representation within field isolate genomes
```{r}
# an absolute quantification of representation by each genome by number of bases represented

Represented <- rbind(Cterm %>% mutate(domain = "C-term"), Nterm %>% mutate(domain = "N-term")) %>% group_by(domain, genome) %>% summarise(totals = sum(domain_length), aligned  = sum(aligned)) %>% mutate(proportion = aligned / totals)


#exclude tbgDAL972 since the assembly does not include complete characterization of VSG encoding regions
Represented <- Represented[!Represented$genome == "tbgDAL972", ]

Represented <- merge(Represented, genomeMeta) %>% 
  mutate(groups = case_when(year < "1999" ~ "Cote d'Ivoire 1980's",
                            year > "2004" ~ "DRC 2000's",
                            TRUE ~ "Cote d'Ivoire 2000's"))

Rep_summary <- Represented %>% group_by(domain, groups) %>% summarise(stddev = sd(proportion), proportion = mean(proportion))

ggplot(Represented[Represented$domain == "N-term", ], aes(x = groups, y = proportion, fill = groups)) + geom_dotplot(binaxis = "y", stackdir = "center", dotsize = 0.8, position = position_dodge(1)) +
  theme_bw() +
  scale_fill_viridis(discrete = T) + 
  ggtitle("N-terminal Domain Representation") +
  ylab("Nucleotides Aligned / Total VSG Nucleotides") +
  xlab("") + 
  theme(legend.position = "none") + 
  geom_crossbar(data = Rep_summary[Rep_summary$domain == "N-term", ], 
                aes(ymin = proportion, ymax = proportion), size = 0.3, color = "black") + 
  geom_errorbar(data = Rep_summary[Rep_summary$domain == "N-term", ], 
                aes(ymin = (proportion - stddev), ymax = (proportion + stddev)), width = 0.3, color = "black") + scale_y_continuous(limits = c(0,0.2))


ggplot(Represented[Represented$domain == "C-term", ], aes(x = groups, y = proportion, fill = groups)) + geom_dotplot(binaxis = "y", stackdir = "center", dotsize = 0.8, position = position_dodge(1)) +
  theme_bw() +
  scale_fill_viridis(discrete = T) + 
  ggtitle("C-terminal Domain Representation") +
  ylab("Nucleotides Aligned / Total VSG Nucleotides") +
  xlab("") + 
  theme(legend.position = "none") + 
  geom_crossbar(data = Rep_summary[Rep_summary$domain == "C-term", ], 
                aes(ymin = proportion, ymax = proportion), size = 0.3, color = "black") + 
  geom_errorbar(data = Rep_summary[Rep_summary$domain == "C-term", ], 
                aes(ymin = (proportion - stddev), ymax = (proportion + stddev)), width = 0.3, color = "black") + scale_y_continuous(limits = c(0.5,0.85))

# kruskal wallis test: nonparametric, independent groups, no repeated measures
# post hoc dunn test. determine which groups are significantly different from each other

# N-terms
kruskal_test(Represented[Represented$domain == "N-term", ], proportion ~ groups)
dunn_test(Represented[Represented$domain == "N-term", ], proportion ~ groups, p.adjust.method = "holm")

# C-terms
kruskal_test(Represented[Represented$domain == "C-term", ], proportion ~ groups)
dunn_test(Represented[Represented$domain == "C-term", ], proportion ~ groups, p.adjust.method = "holm")

```


### Fig5C: Determine where sequences of homology fall on phyre2 predicted VSG structures
```{r}
#read phyre2 output metadata file
##these are the top hits
phyre2 <- read_delim(file = "FigureData/Phyre2_summaryinfo.txt", delim = "|")
colnames(phyre2)[1] <- "VSG"
# remove whitespace
phyre2$VSG <- gsub(" ", "", phyre2$VSG)
# join by cluster ref
phyre2 <- inner_join(phyre2, cluster_ref, by = "VSG")

# find regions of no homology, 0 alignments by any genome
noSimilarity <- data.frame(start = double(),
                     end = double(),
                     width = double(),
                     VSG = character())
for (i in 1:length(unique(plot30bp$VSG))) {
    irR <- IRanges(plot30bp %>%
                   .[.$VSG == unique(plot30bp$VSG)[i], ] %>%
                   select(VSG, start, end) %>%
                   distinct() %>%
                   .$start,
                 end = plot30bp %>%
                   .[.$VSG == unique(plot30bp$VSG)[i], ] %>%
                   select(VSG, start, end) %>%
                   distinct() %>%
                   .$end)
    # collapse overlapping regions to calculate VSG coverage
    cov <- IRanges::reduce(irR) %>% as.data.frame()
    cov <- cov %>% mutate(VSG = unique(plot30bp$VSG)[i])
    noSimilarity <- rbind(noSimilarity, cov)
}

noSimilarity <- inner_join(noSimilarity, phyre2, by = "VSG") %>% 
    merge(plot30bp %>% select(cluster, seqlen, nterm_coord) %>% distinct())

# regions of no similarity for all VSG
noSimilarity %>% ggplot() +
  geom_rect(aes(xmin = start, xmax = end,
                       ymin = 0, ymax = 1)) +
  scale_fill_viridis(discrete = T) +
  facet_wrap(~cluster) +
  xlab("Position (bp)") +
  ggtitle("30bp query") +
  theme_bw() +
  geom_vline(aes(xintercept = seqlen), linetype = "solid", color = "blue", size = 0.6) +
  geom_vline(aes(xintercept = nterm_coord), linetype = "solid", color = "red", size = 0.6)

# Gambiense 248: type B with highest overall coverage
noSimilarity[noSimilarity$cluster == "Gambiense 248", ] %>% ggplot() +
  geom_rect(aes(xmin = start, xmax = end,
                       ymin = 0, ymax = 1)) +
  scale_fill_viridis(discrete = T) +
  facet_wrap(~cluster, nrow = 1) +
  xlab("Position (bp)") +
  theme_bw() +
  geom_vline(aes(xintercept = seqlen), linetype = "solid", color = "royalblue", size = 0.6) +
  geom_vline(aes(xintercept = nterm_coord), linetype = "solid", color = "palegreen", size = 0.6)

# Gambiense 400: type A with highest overall coverage
noSimilarity[noSimilarity$cluster == "Gambiense 400", ] %>% ggplot() +
  geom_rect(aes(xmin = start, xmax = end,
                       ymin = 0, ymax = 1)) +
  scale_fill_viridis(discrete = T) +
  facet_wrap(~cluster, nrow = 1) +
  xlab("Position (bp)") +
  theme_bw() +
  geom_vline(aes(xintercept = seqlen), linetype = "solid", color = "royalblue", size = 0.6) +
  geom_vline(aes(xintercept = nterm_coord), linetype = "solid", color = "palegreen", size = 0.6) + scale_x_continuous(limits = c(0,1500))

# Gambiense 452: type A VSG of around average coverage
noSimilarity[noSimilarity$cluster == "Gambiense 452", ] %>% ggplot() +
  geom_rect(aes(xmin = start, xmax = end,
                       ymin = 0, ymax = 1)) +
  scale_fill_viridis(discrete = T) +
  facet_wrap(~cluster, nrow = 1) +
  xlab("Position (bp)") +
  theme_bw() +
  geom_vline(aes(xintercept = seqlen), linetype = "solid", color = "royalblue", size = 0.6) +
  geom_vline(aes(xintercept = nterm_coord), linetype = "solid", color = "palegreen", size = 0.6) +
  scale_x_continuous(limits = c(0,1500))

Similarity_cov <- noSimilarity %>% group_by(cluster) %>% 
    summarize(aligned = sum(width)) %>% 
    full_join(bt30bp %>% select(cluster, seqlen) %>% distinct()) %>%  
    mutate(pct_coverage = (aligned / seqlen)*100)
# pct_coverage for each VSG by all datasets
Similarity_cov

```


### Supplemental Figure 8A
```{r}

# distribution plots of coverage over full VSG ORF
dist20bp <- calc_cov_20bp %>% 
  inner_join(genomeMeta) %>% 
  mutate(dist_facets = case_when(year < "1999" ~ "Cote d'Ivoire 1980's",
                                 year > "2004" ~ "DRC 2000's",
                                 TRUE ~ "Cote d'Ivoire 2000's")) %>% 
  mutate(query_size = "20bp")

dist30bp <- calc_cov_30bp %>% 
  inner_join(genomeMeta) %>% 
  mutate(dist_facets = case_when(year < "1999" ~ "Cote d'Ivoire 1980's",
                                 year > "2004" ~ "DRC 2000's",
                                 TRUE ~ "Cote d'Ivoire 2000's")) %>% 
  mutate(query_size = "30bp")

dist50bp <- calc_cov_50bp %>% 
  inner_join(genomeMeta) %>% 
  mutate(dist_facets = case_when(year < "1999" ~ "Cote d'Ivoire 1980's",
                                 year > "2004" ~ "DRC 2000's",
                                 TRUE ~ "Cote d'Ivoire 2000's")) %>% 
  mutate(query_size = "50bp")

# exclude tbgDAL972 it is an outlier
# the assembly has inadequate VSG representation
g1 <- ggplot(data = rbind(dist20bp[dist20bp$dist_facets == "Cote d'Ivoire 1980's", ], dist30bp[dist30bp$dist_facets == "Cote d'Ivoire 1980's", ], dist50bp[dist50bp$dist_facets == "Cote d'Ivoire 1980's", ]) %>% .[!.$genome == "tbgDAL972", ],
       aes(x = pct_coverage, fill = genome)) +
  geom_density(alpha = 0.4) +
  scale_color_viridis(discrete = T) +
  scale_fill_viridis(discrete = T) +
  xlab("Percent Patient VSG Gene Coverage") +
  ggtitle("Cote d'Ivoire 1980's") +
  facet_wrap(~query_size, ncol = 1, scales = "free_y") +
  scale_x_continuous(limits = c(0,100)) +
  theme_bw() 

g2 <- ggplot(data = rbind(dist20bp[dist20bp$dist_facets == "Cote d'Ivoire 2000's", ], dist30bp[dist30bp$dist_facets == "Cote d'Ivoire 2000's", ], dist50bp[dist50bp$dist_facets == "Cote d'Ivoire 2000's", ]),
       aes(x = pct_coverage, fill = genome)) +
  geom_density(alpha = 0.4) +
  scale_color_viridis(discrete = T) +
  scale_fill_viridis(discrete = T) +
  xlab("Percent Patient VSG Gene Coverage") +
  ggtitle("Cote d'Ivoire 2000's") +
  facet_wrap(~query_size, ncol = 1, scales = "free_y") +
  scale_x_continuous(limits = c(0,100)) +
  theme_bw() 

g3sum <- rbind(dist20bp[dist20bp$dist_facets == "DRC 2000's", ], dist30bp[dist30bp$dist_facets == "DRC 2000's", ], dist50bp[dist50bp$dist_facets == "DRC 2000's", ]) %>% group_by(query_size, genome) %>% summarize(pct_coverage = mean(pct_coverage))
g3 <- ggplot(data = rbind(dist20bp[dist20bp$dist_facets == "DRC 2000's", ], dist30bp[dist30bp$dist_facets == "DRC 2000's", ], dist50bp[dist50bp$dist_facets == "DRC 2000's", ]),
       aes(x = pct_coverage, fill = genome)) +
#  geom_vline(data = g3sum, aes(xintercept = pct_coverage, color = genome), linetype = "dashed") +
  geom_density(alpha = 0.4) +
#  scale_color_viridis(discrete = T) +
  scale_fill_viridis(discrete = T) +
  xlab("Percent Patient VSG Gene Coverage") +
  ggtitle("DRC 2000's") +
  facet_wrap(~query_size, ncol = 1, scales = "free_y") +
  scale_x_continuous(limits = c(0,100)) +
  theme_bw() 

plot_grid(g1, g2, g3, ncol = 3)

```


### Supplemental Figure 8B
```{r}
# make plot of ranges by genome group. Collapse ranges for all genomes isolated in same year/region

CDI1980 <- plot30bp[plot30bp$year < "1999" & !plot30bp$genome == "tbgDAL972", ]
unique(CDI1980$genome)
CDI2000 <- plot30bp[plot30bp$year > "1999" & plot30bp$year < "2005", ]
unique(CDI2000$genome)
DRC2000 <- plot30bp[plot30bp$year > "2004", ]
unique(DRC2000$genome)

CDI1980.noSimilarity <- data.frame(start = double(),
                     end = double(),
                     width = double(),
                     VSG = character())
for (i in 1:length(unique(CDI1980$VSG))) {
    irR <- IRanges(CDI1980 %>%
                   .[.$VSG == unique(CDI1980$VSG)[i], ] %>%
                   select(VSG, start, end) %>%
                   distinct() %>%
                   .$start,
                 end = CDI1980 %>%
                   .[.$VSG == unique(CDI1980$VSG)[i], ] %>%
                   select(VSG, start, end) %>%
                   distinct() %>%
                   .$end)
    # collapse overlapping regions to calculate VSG coverage
    cov <- IRanges::reduce(irR) %>% as.data.frame()
    cov <- cov %>% mutate(VSG = unique(CDI1980$VSG)[i])
    CDI1980.noSimilarity <- rbind(CDI1980.noSimilarity, cov)
}

CDI2000.noSimilarity <- data.frame(start = double(),
                     end = double(),
                     width = double(),
                     VSG = character())
for (i in 1:length(unique(CDI2000$VSG))) {
    irR <- IRanges(CDI2000 %>%
                   .[.$VSG == unique(CDI2000$VSG)[i], ] %>%
                   select(VSG, start, end) %>%
                   distinct() %>%
                   .$start,
                 end = CDI2000 %>%
                   .[.$VSG == unique(CDI2000$VSG)[i], ] %>%
                   select(VSG, start, end) %>%
                   distinct() %>%
                   .$end)
    # collapse overlapping regions to calculate VSG coverage
    cov <- IRanges::reduce(irR) %>% as.data.frame()
    cov <- cov %>% mutate(VSG = unique(CDI2000$VSG)[i])
    CDI2000.noSimilarity <- rbind(CDI2000.noSimilarity, cov)
}

DRC2000.noSimilarity <- data.frame(start = double(),
                     end = double(),
                     width = double(),
                     VSG = character())
for (i in 1:length(unique(DRC2000$VSG))) {
    irR <- IRanges(DRC2000 %>%
                   .[.$VSG == unique(DRC2000$VSG)[i], ] %>%
                   select(VSG, start, end) %>%
                   distinct() %>%
                   .$start,
                 end = DRC2000 %>%
                   .[.$VSG == unique(DRC2000$VSG)[i], ] %>%
                   select(VSG, start, end) %>%
                   distinct() %>%
                   .$end)
    # collapse overlapping regions to calculate VSG coverage
    cov <- IRanges::reduce(irR) %>% as.data.frame()
    cov <- cov %>% mutate(VSG = unique(DRC2000$VSG)[i])
    DRC2000.noSimilarity <- rbind(DRC2000.noSimilarity, cov)
}

GroupRanges <- rbind(CDI1980.noSimilarity %>% 
                       mutate(group = "Cote d'Ivoire 1980's"), 
                     CDI2000.noSimilarity %>% 
                       mutate(group = "Cote d'Ivoire 2000's"), 
                     DRC2000.noSimilarity %>% mutate(group = "DRC 2000's")) %>%
  mutate(plotbins = as.factor(.$group) %>% as.numeric()) %>% 
  inner_join(phyre2, by = "VSG") %>% 
  merge(plot30bp %>% select(cluster, nterm_coord, seqlen) %>% unique())


GroupRanges %>% ggplot() +
  geom_rect(aes(xmin = start, xmax = end,
                       ymin = plotbins, ymax = plotbins + 0.9,
                fill = group), show.legend = T) +
  scale_fill_viridis(discrete = T) +
  geom_vline(aes(xintercept = seqlen), linetype = "solid", color = "blue", size = 0.6) +
  geom_vline(aes(xintercept = nterm_coord), linetype = "solid", color = "red", size = 0.6) +
  facet_wrap(~cluster) +
  xlab("Position (bp)") +
  ggtitle("Position and Range of Bowtie Alignments: 30bp query < 2 mismatches") +
  theme_bw()
  
```

