---
title: "figure 4"
author: "Jaime So"
date: "8/30/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
library(tidyverse)
library(reshape2)
library(ggsci)
library(xlsx)
```

#Generate Plots for Figure 4: 
##VSG N-terminal type composition fluctuates over the course of infection

###import mouse VSG seq data
Data obtained from paper supplement
* https://science.sciencemag.org/content/suppl/2015/03/25/347.6229.1470.DC1

Nucleic acid fasta files for this experiment available on GitHub:
https://github.com/mugnierlab/VSGSeqPipeline/tree/master/Science2015_reference_genomes/
* VSG fasta broken into early and late
* early refers to 6-30 days post infection and includes VSG sequences assembled in mice 1, 2, 3, 4
* late refers to 96-105 d.p.i. and only mouse 3 had sequencing data for this time period
* early and late VSG fasta files concatenated and translated using biopython
* merge find_VSG_domains.py output with VSG-seq output by VSG ID


```{r}

# import merged table of VSGseq and N-terminal typing data 
# make a new column that is general type rather than subtype
allmice_typed <- read_csv("FigureData/Mugnier2015_allMiceTyped.csv") %>% mutate(nterm_type = sub("[[:digit:]]", "", .$nterm_typesubtype))
PlotMouse <- allmice_typed[complete.cases(allmice_typed), ] %>% group_by(mouse, day, nterm_type) %>% summarise(n = n())

PlotMouse <- merge(PlotMouse, PlotMouse %>% group_by(mouse, day) %>% summarise(total = sum(n)))
PlotMouse$pct <- (PlotMouse$n / PlotMouse$total)*100
PlotMouse$day <- as.character(PlotMouse$day)
PlotMouse$day <- as.numeric(PlotMouse$day)

```

##Fig. 4A
###Plot of the early infection (6-30 d.p.i.): number of expressed types over time
```{r}

Fig4A <- ggplot() +
  geom_line(data = PlotMouse[PlotMouse$day < 95, ], aes(x = day, y = pct, color = nterm_type), size = 1) +
  geom_line(data = PlotMouse[PlotMouse$day < 95, ], aes(x = day, y = total, group = mouse), linetype = "twodash") +
  scale_y_continuous(name = "% A VSG or % B VSG", limits = c(0, 100), sec.axis =  sec_axis(~./1, name = "Total Number of Detected VSG")) + 
  facet_wrap(~mouse) +
  scale_color_manual(values = c("steelblue1", "tomato"))+
  theme_bw()
Fig4A

```

##Fig. 4B
###Plot of the late infection (96-105 d.p.i): number of expressed types over time
```{r}

Fig4B <- ggplot() +
  geom_line(data = PlotMouse[PlotMouse$day > 95, ], aes(x = day, y = pct, color = nterm_type), size = 1) +
  geom_line(data = PlotMouse[PlotMouse$day > 95, ], aes(x = day, y = total, group = mouse), linetype = "twodash") +
  scale_y_continuous(name = "% A VSG or % B VSG", limits = c(0, 100), sec.axis =  sec_axis(~./1, name = "Total Number of Detected VSG")) + 
  facet_wrap(~mouse) +
  scale_color_manual(values = c("steelblue1", "tomato"))+
  scale_x_continuous(breaks = c(96, 99, 102, 105))+
  theme_bw()
Fig4B

```

##Supplemental Figure 2
###Does parasitemia correlate with VSG diversity?
```{r}
#load up exported mouse data from Mugnier 2015 Science paper
# For the Mugnier mouse data, the parasitemia for each animal on various days is included in the supplemental data files that contain VSG-seq output data
# the VSG is listed as 'parasitemia'

# I have the parasitemia for each mouse by day and the expressed VSG subtype data
# although in different dataframes

#find the total number of expressed VSGs for each day
totalMouseVSG <- allmice_typed %>% group_by(mouse, day, nterm_type) %>% summarise(VSG = n()) %>% .[complete.cases(.),]
# not all of the days that parasitemia was measured were also sequenced
# I only want to consider days that have both parasitemia data and sequenced VSGs to see if there is a correlation between parasitemia and the number of unique detectable VSG

# need to join by both mouse and day, make a new column with both
parasitemia <- read_csv("FigureData/Mugnier2015_parasitemia.csv")
parasitemia$ugh <- c(paste(parasitemia$mouse, parasitemia$day, sep = ","))
totalMouseVSG$ugh <- c(paste(totalMouseVSG$mouse, totalMouseVSG$day, sep = ","))
totalMouseVSG <- totalMouseVSG %>% group_by(ugh) %>% summarise(total = sum(VSG)) %>% inner_join(totalMouseVSG)
plotMouseCorr <- inner_join(parasitemia, totalMouseVSG, by = 'ugh')

#clean up the dataframe by getting rid of duplicate columns and redundant data
plotMouseCorr <- plotMouseCorr[,c(3,4,5,8,11,12)]
colnames(plotMouseCorr) <- c("day", "parasites", "mouse", "totalVSG", "nterm", "numType")

# first find the correlation coefficient for total VSG diversity vs. parasitemia
cleanMouseCorr <- unique(plotMouseCorr[ ,(1:4)])
# will add the coefficient to the plot as geom_text
# make a small dataframe that includes the coefficient and my desired plot coordinates
# calculate the pearson correlation coefficient
mouse.cor.coeff <- data.frame(cor = as.character(round(cor(cleanMouseCorr[ , "totalVSG"], cleanMouseCorr[ , "parasites"], method = "pearson"), digits = 3)),
                      x = c(80),
                      y = c(1e+9))

# parasitemia vs Number of VSGs (for each replicate)
VSGs1 <- cleanMouseCorr %>%
  ggplot(aes(x=totalVSG, y=parasites))+
  geom_point(stat = "identity")+
  geom_smooth(method = lm, color = "black", se = FALSE)+
  xlab("Number of Unique VSGs")+
  ylab("parasites/mL blood")+
  geom_text(data=mouse.cor.coeff, aes(label=cor, x=x, y=y))+
  theme_classic()
VSGs1

# parasitemia vs. n-terminal type
typeCorr <- plotMouseCorr %>% group_by(day, parasites, mouse, nterm) %>% summarise(pct = (numType / totalVSG)*100)

# make a small dataframe that includes the coefficient and my desired plot coordinates
# calculate the pearson correlation coefficient
A.cor.coeff <- data.frame(cor = as.character(round(cor(typeCorr[typeCorr$nterm == "A" , ] %>% .$pct , typeCorr[typeCorr$nterm == "A" , ] %>% .$parasites, method = "pearson"), digits = 3)),
                      x = c(80),
                      y = c(1e+9))

VSGsA <- typeCorr[typeCorr$nterm == "A",] %>%
  ggplot(aes(x=pct, y=parasites))+
  geom_point(stat = "identity")+
  geom_smooth(method = lm, color = "black", se = FALSE)+
  xlab("Percent Type A VSG")+
  ylab("parasites/mL blood")+
  geom_text(data=A.cor.coeff, aes(label=cor, x=x, y=y))+
  theme_classic()
VSGsA

B.cor.coeff <- data.frame(cor = as.character(round(cor(typeCorr[typeCorr$nterm == "B" , ] %>% .$pct , typeCorr[typeCorr$nterm == "B" , ] %>% .$parasites, method = "pearson"), digits = 3)),
                      x = c(80),
                      y = c(1e+9))

VSGsB <- typeCorr[typeCorr$nterm == "B",] %>%
  ggplot(aes(x=pct, y=parasites))+
  geom_point(stat = "identity")+
  geom_smooth(method = lm, color = "black", se = FALSE)+
  xlab("Percent Type B VSG")+
  ylab("parasites/mL blood")+
  geom_text(data=B.cor.coeff, aes(label=cor, x=x, y=y))+
  theme_classic()
VSGsB

```

